### PR TITLE
Refactor on Generics to use Async Await

### PR DESCRIPTION
**Pull request by borbe**
Refactor on Generics/action-steps.js & Generics/verification-steps.js to use Async Await

### CODE REVIEW

File: generics/action-steps.js [hunk 1]
```diff
@@ -1,140 +1,48 @@
...
+    When(/^user presses the (.*) key/, async (key) => {
+        await driver.actions().sendKeys(webdriver.Key[key]).perform();
     });
 
-    Then(/^user waits for ([0-9]+) seconds$/, async (num) => {
+    When(/^user waits for ([0-9]+) seconds$/, async (num) => {
         await driver.sleep(num * 1000);
     });
-    //
-    // Then(/^user waits for "([^"]*)" element to be present$/, { timeout: 2 * 60 * 1000 }, (elementText, next) => {
```
Comment (borbe) [inline generics/action-steps.js:80]: go rid of the commented code that seems to be not useful anymore.

let me know if i should keep any of it, my personal asses is to not keep it, most likely we will not need it in the future as we have not had any major issues in the executions that used abacus.

File: generics/verification-steps.js [hunk 2]
```diff
@@ -1,183 +1,71 @@
+import {expect} from 'chai';
+
 module.exports = ({Then}) => {
 
     Then(/^user is taken to the "([^"]*)" page$/, async (pageURI) => {
         global.pageID = pageURI;
-
         if (!pageMap[global.pageID]) throw new Error(`Page Object with name ${pageURI} is not defined`);
-        const result = await driver.wait(until.urlMatches(pageMap[global.pageID].URL), process.env.CUCUMBER_REDIRECT_TIMEOUT * 1000, `URL never matched ${pageMap[global.pageID].URL}`)
```
Comment (borbe) [inline generics/verification-steps.js]: got rid of this but it was commented from you (seems like a newly refactored function), please take a look at this one. @g2valdez

File: generics/verification-steps.js [hunk 3]
```diff
@@ -1,183 +1,71 @@
...
+        await pageMap[global.pageID][labelID].waitUntil[valOrText][matchType](text);
+        expect(pageMap[global.pageID][labelID].getAttribute(valOrText)).to.eventually.contain(text);
     });
 
+    Then(/^the "([^"]*)" element (value|text) (is not|does not contain|does not match) "(.*)"$/, async (labelText, valOrText, matchType, text) => {
+        const labelID = labelText.replace(/ /g, '_').toUpperCase();
+        expect(pageMap[global.pageID][labelID].getAttribute(valOrText)).to.eventually.not.contain(text);
+    });
 
-    // Then(/^the "([^"]*)" element value does not contain "([^"]*)"$/, (buttonText, text, next) => {
```
Comment (borbe) [inline generics/verification-steps.js:20]: got rid of these stuff as i implemented the function in line 16

File: generics/verification-steps.js [hunk 4]
```diff
@@ -1,183 +1,71 @@
...
-          });
+        if (yorn === 'is not') {
+            if (result === true) throw new Error(`${elementID} Element is displayed`);
+        }
+        else {
+            if (result !== true) throw new Error(`${elementID} Element is not displayed`);
+        }
     });
 
-    // const validateImages = function (webTitle) {
```
Comment (borbe) [inline generics/verification-steps.js:143]: got rid of image validator, if we implement this in the near future, most likely we will do it in a different manner.
as it was flaky a t m

File: generics/verification-steps.js (hunk 2 above)
Comment (willievaldez) [reply generics/verification-steps.js]: you should revert this one. I uncommented the remaining code in the function, as it now works

**Comment by borbe** [comment]
@g2valdez done, please check

File: generics/verification-steps.js [hunk 5]
```diff
@@ -1,183 +1,78 @@
+import {expect} from 'chai';
```
Comment (willievaldez) [inline generics/verification-steps.js]: No imports please

**Review by willievaldez — APPROVED** [review]
beautiful

**Pull request merged**